tofhir-engine/src/main/scala/io/tofhir/engine/data/read/FileDataSourceReader.scala

```diff
@@ -35,18 +35,29 @@ class FileDataSourceReader(spark: SparkSession) extends BaseDataSourceReader[Fil
     var rowIndex: Integer = 0
     //Based on source type
     val resultDf = mappingSource.sourceType match {
-        case SourceFileFormats.CSV =>
+        case SourceFileFormats.CSV | SourceFileFormats.TSV =>
+          val updatedOptions = mappingSource.sourceType match {
+            case SourceFileFormats.TSV =>
+              // If the file format is tsv, use tab (\t) as separator by default if it is not set explicitly
+              mappingSource.options +
+                ("sep" -> mappingSource.options.getOrElse("sep", "\\t"),
+                // use *.tsv as pathGlobFilter by default if it is not set explicitly to ignore files without tsv extension
+                "pathGlobFilter" -> mappingSource.options.getOrElse("pathGlobFilter", s"*.${SourceFileFormats.TSV}"))
+            case SourceFileFormats.CSV =>
+              mappingSource.options +
+                // use *.csv as pathGlobFilter by default if it is not set explicitly to ignore files without csv extension
+                ("pathGlobFilter" -> mappingSource.options.getOrElse("pathGlobFilter", s"*.${SourceFileFormats.CSV}"))
+          }
+
           //Options that we infer for csv
           val inferSchema = schema.isEmpty || mappingSource.preprocessSql.isDefined
           val csvSchema = if(mappingSource.preprocessSql.isDefined) None else schema
           //val enforceSchema = schema.isDefined
           val includeHeader = mappingSource.options.get("header").forall(_ == "true")
-          val fileFilter = mappingSource.options.getOrElse("pathGlobFilter", "*.csv")
```

Comment:
Does this mean that we are now processing all files under a given folder without any extension filter? It is going to be the user's responsibility to put the files in the correct format w.r.t his/her configuration, am I right?

What will happen if the reader encounters a txt file with some dummy text in it?

Reply:
Actually no. Previous limitations remain. We just moved pathGlobFilter param to otherOptions map. And used it in spark.read section. Giving it to the spark explicitly (first one) would be same with the second one if we put pathGlobFilter param in otherOptions.

```
   spark.readStream
          .option("pathGlobFilter",fileFilter) // Ignore files without csv extension in default
          .option("enforceSchema", false) //Enforce schema should be false (See https://spark.apache.org/docs/latest/sql-data-sources-csv.html)
          .option("header", includeHeader)
          .option("inferSchema", inferSchema)
          .options(otherOptions)
          .schema(csvSchema.orNull)
          .csv(finalPath)
```

   ```
     spark.readStream
          .option("enforceSchema", false) //Enforce schema should be false (See https://spark.apache.org/docs/latest/sql-data-sources-csv.html)
          .option("header", includeHeader)
          .option("inferSchema", inferSchema)
          .options(otherOptions)
          .schema(csvSchema.orNull)
          .csv(finalPath)
```